app/services/braintree_services/payment_options.rb

```diff
@@ -4,12 +4,34 @@ module BraintreeServices
   class PaymentOptions
     attr_reader :params
 
-    def initialize(params)
+    def initialize(params, cookied_payment_methods)
       @params = params
+      @cookied_payment_methods = cookied_payment_methods
     end
 
     def payment_method
       @payment_method ||= customer.payment_methods.find(params[:payment][:payment_method_id])
+
+      raise PaymentProcessor::Exceptions::PaymentMethodNotFound unless @payment_method
+      @payment_method
+    end
+
+    def valid_payment_method_id
+      cookied_payment_methods.split(',').include?(params[:payment][:payment_method_id])
+    end
+
+    def payment_method_id
+      return nil unless valid_payment_method_id
+      return nil unless member&.customer
```

Comment:
we're now using this in two places, so should we move this to a member model concern?